packages/ra-ui-materialui/src/field/ReferenceArrayField.tsx

```diff
@@ -80,7 +103,22 @@ const useStyles = makeStyles(
     { name: 'RaReferenceArrayField' }
 );
 
-export const ReferenceArrayFieldView = ({
+type ReferenceArrayFieldClassKey = 'progress';
```

Comment:
Where does this come from?

Reply:
This is from key of the [`makeStyle` object in line 78](https://github.com/marmelab/react-admin/pull/4448/files/36635530bed3b6d66f1888d1b19715b64a3a6bf7#diff-edf706d49e4f05c200bd721a16f0334bR101) and it's similar to what material-ui is doing with its components
e.g. [`AppBar`](https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/AppBar/AppBar.d.ts#L17-L26). It'd allow library user to know which classes this component they can override

Reply:
[`Placeholder.tsx`](https://github.com/marmelab/react-admin/blob/master/packages/ra-ui-materialui/src/list/Placeholder.tsx#L17) is using `Record<'root', string>` for `classes` prop,
while `ClassNameMap` is string a type alias of `Record` that sets default type to `string` from material-ui
```typescript
ClassNameMap<ClassKey extends string = string> = Record<ClassKey, string>
```

Reply:
I name it `ReferenceArrayFieldClassKey` so that I can reuse it in [line 110](https://github.com/marmelab/react-admin/pull/4448/files/36635530bed3b6d66f1888d1b19715b64a3a6bf7#diff-edf706d49e4f05c200bd721a16f0334bR110)